### PR TITLE
Split openshift authorizer interface to match upstream

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/13157

Builds on https://github.com/openshift/origin/pull/13286

Upstream has an authorizer interface and a separate method for locating subjects.  This makes the same split in our code and removes our authorizer.Authorizer interface entirely.

This is the last commit in my chain.

@liggitt @enj

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

File: pkg/cmd/server/origin/master_config.go [hunk 1]
```diff
@@ -817,8 +817,9 @@ func newAuthorizationAttributeBuilder(requestContextMapper kapi.RequestContextMa
 		sets.NewString(bootstrappolicy.AuthenticatedGroup),
 		requestInfoFactory,
 	)
+	personalSARRequestInfoResolveer := authorizer.NewPersonalSARRequestInfoResolver(browserSafeRequestInfoResolver)
```
Comment (sttts) [inline pkg/cmd/server/origin/master_config.go]: -e

File: pkg/authorization/authorizer/cache/authorizer.go [hunk 2]
```diff
@@ -91,37 +90,6 @@ func (c *CacheAuthorizer) Authorize(a kauthorizer.Attributes) (allowed bool, rea
 	return allowed, reason, err
 }
 
-func (c *CacheAuthorizer) GetAllowedSubjects(attributes kauthorizer.Attributes) (sets.String, sets.String, error) {
```
Comment (ncdc) [inline pkg/authorization/authorizer/cache/authorizer.go]: I know you've split the interface in 2. Does the code that is using the `CacheAuthorizer` not ever call `GetAllowedSubjects`?

File: pkg/authorization/authorizer/cache/authorizer.go [hunk 3]
```diff
@@ -91,37 +90,6 @@ func (c *CacheAuthorizer) Authorize(a kauthorizer.Attributes) (allowed bool, rea
...
-	key, err := cacheKey(attributes)
-	if err != nil {
-		glog.V(5).Infof("could not build cache key for %#v: %v", attributes, err)
-		return c.authorizer.GetAllowedSubjects(attributes)
-	}
-
-	if value, hit := c.allowedSubjectsCache.Get(key); hit {
-		switch record := value.(type) {
-		case *allowedSubjectsCacheRecord:
```
Comment (ncdc) [inline pkg/authorization/authorizer/cache/authorizer.go]: Remove the `allowedSubjectsCacheRecord` struct definition too?

**Comment by ncdc** [comment]
@deads2k I was trying to run `hack/test-cmd.sh` and it isn't able to log in:

```
++ /home/vagrant/go/src/github.com/openshift/origin/test/cmd/annotations.sh
FAILURE after 0.204s: hack/test-cmd.sh:223: executing 'oc login --server='https://127.0.0.1:28443' --certificate-authority='/tmp/openshift/test-cmd//openshift.local.config/master/ca.crt' -u test-user -p anything' expecting success: the command returned the wrong error code
Standard output from the command:
Login successful.


Standard error from the command:
Error from server (Forbidden): User "test-user" cannot "create" on ""
```

File: pkg/authorization/authorizer/cache/authorizer.go (hunk 2 above)
Comment (deads2k) [inline pkg/authorization/authorizer/cache/authorizer.go]: > I know you've split the interface in 2. Does the code that is using the CacheAuthorizer not ever call GetAllowedSubjects?

Yeah, funny that, right?  Turns out we hardly ever use it.

**Comment by enj** [comment]
> Error from server (Forbidden): User "test-user" cannot "create" on ""

See https://github.com/openshift/origin/pull/13286#discussion_r105067842

**Comment by deads2k** [comment]
> @deads2k I was trying to run hack/test-cmd.sh and it isn't able to log in:

I'd gotten used to the 10-15 minute CI return result for tests.  It's been really slow dealing with for origin.  not improved by really bad build times for test/integration.  Guess I may have to bit the bullet, but geez it's bad.

File: pkg/cmd/server/kubernetes/node_auth.go [hunk 4]
```diff
@@ -7,17 +7,16 @@ import (
 
 	"github.com/golang/glog"
 
-	kauthorizer "k8s.io/kubernetes/pkg/auth/authorizer"
+	authorizer "k8s.io/kubernetes/pkg/auth/authorizer"
```
Comment (enj) [inline pkg/cmd/server/kubernetes/node_auth.go]: Drop the unneeded name.

**Comment by enj** [comment]
LGTM if tests pass.

**Comment by deads2k** [comment]
re[test]

**Comment by deads2k** [comment]
[merge]

**Comment by sttts** [comment]
❤️

**Pull request merged**